src/types.ts

```diff
@@ -0,0 +1,23 @@
+export interface PaymentReceivedEvent {
+  type: "payment-received";
+  network: string; // https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md
+  amount: string;
+  currency: string;
+  address: string;
+}
```

Comment:
why not use `Components.Schemas.PushNotificationOnchainAddressGotPaid`  from openapi?

this is also useful, if for example, an external service sends callback to a GC. this external service can use openapi definition right away.

Reply:
Basically, the abstraction is this: 

`Blockchain module -> generate events`
`Worker -> convert events into push notifications (based on subscribed addresses, user preferences etc)`

It's possible to share the event object with the push notification object, but then the blockchain module has to set dummy values for things such as `os` or `badge`. I thought that approach is fair enough in GC today, but with a larger set of blockchain modules I wanted to make it slightly cleaner.

Reply:
I totally agree though that the events could also come from an external service, so an OpenAPI definition would give us an automatic internal API for this I guess.